brigade.js

```diff
@@ -22,6 +24,38 @@ function runSuite(e, p) {
     check.run();
 }
 
+function publish(project, version) {
+    var publish = new Job(`${projectName}-publish`, nodeImage);
+    publish.env = {
+        "NPM_TOKEN": project.secrets.npmToken
+    };
+    // The steps to publish include the steps to build, and then some, so we'll
+    // use the build job steps as a starting point.
+    publish.tasks = build().tasks;
+    // If we leave .npmrc at the root of the project with the NPM_TOKEN env var
+    // unset, all yarn commands will fail. Since this env var is populated with
+    // the correct secret ONLY for this one job, we create .npmrc right here,
+    // just in time.
+    publish.tasks.push("echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}' > .npmrc");
+    publish.tasks.push("apt-get update");
+    publish.tasks.push("apt-get install -y jq");
+    publish.tasks.push("cat package.json | jq '.version |= \"" + version + "\"' > package.json.tmp");
+    publish.tasks.push("rm package.json")
+    publish.tasks.push("mv package.json.tmp package.json")
+    publish.tasks.push("npm publish");
+    return publish;
+}
+
+events.on("push", (e, p) => {
+    let matchStr = e.revision.ref.match(releaseTagRegex);
+    if (matchStr) {
+        // This is an official release with a semantically versioned tag
+        let matchTokens = Array.from(matchStr);
+        let version = matchTokens[1];
+        return publish(p, version).run();
+    }
```

Comment:
Idea: I'm wondering if it would be helpful to add an `else { console.log("Ref ${e.revision.ref} does not match expected official release tag regex (${releaseTagRegex}); not releasing."); }` or some such?  Could be helpful when a tag is pushed intending to be released, but just barely misses the regex... say, `0.1.0` sans `v`, etc.?

Reply:
The only trouble with that would be that this handler is for _all_ push events, which would include not only new tags, but also merges to master. With merges to master occurring more frequently than releases, I'm not sure I'd want to output something on every such occurrence that's basically asking, "Did you mean to release and you just got the tag wrong?"

I am still open to improving this _somehow_ in a follow-up. For now I am going to merge so that we have this working to our benefit with the upcoming 1.2 release.

Reply:
Yea, good point... maybe we can add a check if ref is prefixed with `/refs/tags`... anyway, agreed, we'll brainstorm on this and iterate.

Reply:
That should work.